src/main/java/com/develocity/api/tests/GitHubCliReporter.java

```diff
@@ -0,0 +1,104 @@
+package com.develocity.api.tests;
+
+import com.gradle.enterprise.api.model.TestOrContainer;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.addAll;
+
+final class GitHubCliReporter implements UnstableTestContainersReporter {
+
+    private final String serverUrl;
+    private final String githubRepoUrl;
+    private final OffsetDateTime now;
+    private final List<TestContainerWithCases> unstableContainers;
+    private final Interval buildTimeRange;
+    private final boolean isDryRunMode;
+
+    GitHubCliReporter(
+        String serverUrl,
+        String githubRepoUrl,
+        OffsetDateTime now,
+        List<TestContainerWithCases> unstableContainers,
+        Interval buildTimeRange
+    ) {
+        this(serverUrl, githubRepoUrl, now, unstableContainers, buildTimeRange, false);
+    }
+
+    GitHubCliReporter(
+        String serverUrl,
+        String githubRepoUrl,
+        OffsetDateTime now,
+        List<TestContainerWithCases> unstableContainers,
+        Interval buildTimeRange,
+        boolean isDryRunMode
+    ) {
+        this.serverUrl = serverUrl;
+        this.githubRepoUrl = githubRepoUrl;
+        this.now = now;
+        this.unstableContainers = unstableContainers;
+        this.buildTimeRange = buildTimeRange;
+        this.isDryRunMode = isDryRunMode;
+    }
+
+    @Override
+    public void report() {
+        unstableContainers.forEach(container -> {
+            List<String> cmd = new ArrayList<>();
+            addAll(cmd, "gh", "issue", "create", "--repo", githubRepoUrl);
+            addAll(cmd, "--title", toIssueTitle(container.getContainer()));
+            addAll(cmd, "--body", toIssueBody(container));
+
+            try {
+                if (isDryRunMode) {
+                    System.out.print(String.join(" ", cmd));
+                } else {
+                    new ProcessBuilder()
+                        .command(cmd)
+                        .inheritIO()
+                        .start()
+                        .waitFor();
+                }
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to create GitHub issues for discovered unstable containers", e);
+            }
+        });
+
+    }
+
+    private String toIssueTitle(TestOrContainer container) {
```

Comment:
Can be static

Reply:
Done in https://github.com/gradle/gradle-enterprise-api-samples/pull/90/commits/ae99e045633aa5e8bd4548bf69848ba087013fa9